mptt/filters.py

```diff
@@ -0,0 +1,116 @@
+from django.conf import settings
+from django.contrib.admin import RelatedFieldListFilter
+from django.contrib.admin.options import IncorrectLookupParameters
+from django.contrib.admin.utils import get_model_from_relation
+from django.core.exceptions import ValidationError
+from django.utils.encoding import smart_text
+from django.utils.html import mark_safe
+from django.utils.translation import ugettext_lazy as _
+from django.utils.translation import get_language_bidi
+from django.db.models.fields.related import ForeignObjectRel, ManyToManyField
+
+MPTT_ADMIN_LEVEL_INDENT = getattr(settings, 'MPTT_ADMIN_LEVEL_INDENT', 10)
+
+
+class TreeRelatedFieldListFilter(RelatedFieldListFilter):
+    """
+    Admin filter class which filters models related to parent model with all it's descendants.
+
+     Usage:
+
+    from mptt.filters import TreeRelatedFieldListFilter
+
+    @admin.register(models.MyModel)
+    class MyModelAdmin(admin.ModelAdmin):
+        model = models.MyModel
+        list_filter = (
+            ('my_related_model', TreeRelatedFieldListFilter),
+                    )
+    """
+    template = 'admin/mptt_filter.html'
+
+    # Ripped from contrib.admin.filters,RelatedFieldListFilter Django 1.8 to
+    # change lookup name to 'inhierarchy'
+    def __init__(self, field, request, params, model, model_admin, field_path):
```

Comment:
Could this method just call super(...) and then override `lookup_kwarg`, rather than copying the rest of the method?